### PR TITLE
storage: make FLAG_PUBLIC and FLAGS_WRITE part of public API

### PR DESCRIPTION
**Pull request by prusnak**
Small refactor of storage

### CODE REVIEW

File: core/embed/extmod/modtrezorconfig/modtrezorconfig.c [hunk 1]
```diff
@@ -212,7 +212,7 @@ STATIC mp_obj_t mod_trezorconfig_set(size_t n_args, const mp_obj_t *args) {
   uint8_t app = trezor_obj_get_uint8(args[0]) & 0x3F;
```
Comment (andrewkozlik) [inline core/embed/extmod/modtrezorconfig/modtrezorconfig.c]: I suppose it would make sense to also add a constant for the flags mask 0x3F.

Comment (prusnak) [reply core/embed/extmod/modtrezorconfig/modtrezorconfig.c]: Do you think having this as `(0xFF - 0xC0)` is a good (comprehensible) idea?

Comment (prusnak) [reply core/embed/extmod/modtrezorconfig/modtrezorconfig.c]: Addressed in 33c9561dc55b362fec4fedc1a69dcbaf85ede705

**Comment by prusnak** [comment]
Rebased and merged as d761a576896c9867c69739e7d94e6d08ed2f3638

**Pull request closed without merging**